modules/samples/connectedcup/component/api/src/main/java/org/coffeeking/api/ConnectedCupManagerServiceImpl.java

```diff
@@ -87,29 +98,42 @@ public Response updateDevice(@PathParam("device_id") String deviceId, @QueryPara
         } catch (DeviceManagementException e) {
             log.error(e.getErrorMessage());
             return Response.status(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()).build();
+        } catch (DeviceAccessAuthorizationException e) {
+            log.error(e.getErrorMessage(), e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()).build();
```

Comment:
This exception is thrown when there is an error occurred when calling the authorization service,eg: plugin or group management initialization failure. So this would raise due to an internal issue. So isn't the response Internal Server Error is the correct one ?